### PR TITLE
Add label background

### PR DESCRIPTION
**Pull request by AntoineSimoulin**
## Context

This PR add the ability to add background color for the label of bounding boxes. In this case, the background label will be the same as the one used for the box frame. The text itself will automatically be switched to black. This should allow better visibility in cases where brightness and darkness prevent from distinguishing regular labels, as mentioned in #8317

## How to use

We add a `fill_labels` parameter in the [`draw_bounding_boxes`](https://github.com/pytorch/vision/blob/main/torchvision/utils.py#L155) function. As illustrated below, this will draw a colored rectangle with the same size of the text rectangle for the text label. By default, the option is deactivated. 

```python
boxes = torch.tensor(
    [[100, 100, 150, 150], [50, 50, 100, 100], [300, 100, 400, 300]],
    dtype=torch.float32,
)

image = torch.ones((3, 500, 500), dtype=torch.uint8) * 255

colors = ["blue", "yellow", "red"]
image = draw_bounding_boxes(
    image, boxes, colors=colors, width=1, fill=False, labels=colors, fill_labels=True
)
plt.imshow(image.numpy().transpose((1, 2, 0)))
plt.axis("off")
plt.show()
```

image

### CODE REVIEW

File: torchvision/utils.py [hunk 1]
```diff
@@ -187,6 +188,7 @@ def draw_bounding_boxes(
         font_size (int): The requested font size in points.
         label_colors (color or list of colors, optional): Colors for the label text.  See the description of the
             `colors` argument for details.  Defaults to the same colors used for the boxes.
+        fill_labels (bool): If `True` fills the label background with specified color.
```
Comment (NicolasHug) [inline torchvision/utils.py:191]:
```suggestion
        fill_labels (bool): If `True` fills the label background with specified color. Default: False.
```

File: torchvision/utils.py [hunk 2]
```diff
@@ -259,7 +261,14 @@ def draw_bounding_boxes(
...
         if label is not None:
-            margin = width + 1
+            box_margin = 1
+            margin = width + box_margin
+            if fill_labels:
+                left, top, right, bottom = draw.textbbox((bbox[0] + margin, bbox[1] + margin), label, font=txt_font)
+                draw.rectangle(
+                    (left - box_margin, top - box_margin, right + box_margin, bottom + box_margin), fill=color
+                )
+                label_color = "black"
```
Comment (NicolasHug) [inline torchvision/utils.py:271]: Do we have any reason to force `label_color` to black? Could we just make `black` the default, but let the user specify it if they wanted to?

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @AntoineSimoulin , this looks good! I made some minor comments below, it'd be also great to have some small test to check the expected behavior, similar to the existing tests we have for `draw_bounding_boxes`

**Comment by AntoineSimoulin** [comment]
@NicolasHug, thanks for the comments. I did the following modifications:
* Let the user choose the background color and default to black otherwise in [2e72bbb](https://github.com/pytorch/vision/pull/9018/commits/2e72bbbfb2d22241440588f4eefb7b2cccc446c8);
* Added some tests in [9f9e320](https://github.com/pytorch/vision/pull/9018/commits/9f9e320e7439b6c6ac55facffc5df9f3fab7f465), similar to the existing tests we have for [`draw_bounding_boxes`](https://github.com/pytorch/vision/blob/main/test/test_utils.py#L120). Please run with  `pytest test/test_utils.py -vvv -k test_draw_boxes_with_coloured_label_backgrounds`.

File: torchvision/utils.py [hunk 3]
```diff
@@ -187,6 +188,7 @@ def draw_bounding_boxes(
         font_size (int): The requested font size in points.
         label_colors (color or list of colors, optional): Colors for the label text.  See the description of the
             `colors` argument for details.  Defaults to the same colors used for the boxes.
+        fill_labels (bool): If `True` fills the label background with specified color. Default: False.
```
Comment (NicolasHug) [inline torchvision/utils.py:191]: Minor suggestion, as I originally was wondering whether this was `colors` or `label_colors` (which is just for  the text)

```suggestion
        fill_labels (bool): If `True` fills the label background with specified box color (from the ``colors`` parameter). Default: False.
```

File: torchvision/utils.py [hunk 4]
```diff
@@ -187,6 +188,7 @@ def draw_bounding_boxes(
         font_size (int): The requested font size in points.
         label_colors (color or list of colors, optional): Colors for the label text.  See the description of the
             `colors` argument for details.  Defaults to the same colors used for the boxes.
```
Comment (NicolasHug) [inline torchvision/utils.py:190]:
```suggestion
            `colors` argument for details.  Defaults to the same colors used for the boxes, or to black if ``fill_labels`` is True.
```

**Review by NicolasHug — APPROVED** [review]
Thank you!
